### DISCUSSION TITLE
Why is there ScoreBeatPreNotesGlyph() but no ScoreBeatPostNotesGlyph()?

### DISCUSSION THREAD

**Discussion by goodgame365** [CoderLine/alphaTab#1705] [Q&A] [answered]
Why is there ScoreBeatPreNotesGlyph() but no ScoreBeatPostNotesGlyph()? I want to add some characters after the note (Note). How can I achieve this?
![未标题-1](https://github.com/user-attachments/assets/21c8f48a-fb09-47ce-8e08-641a67e0583d)
![image](https://github.com/user-attachments/assets/0f8c51c4-66ec-4fc8-aaea-1677b2cee4c0)

**Comment by Danielku15** [comment, marked as answer]
It is simply the design of how things are organized in alphaTab. From the overall engraving algorithms we have, there is no "logical" place of a "post note".

All annotation and effect glyphs we have are aligned with the note heads or dynamically placed (e.g. ties). 
They are then usually rather part of the "BeatOnNote" (like the dots after note heads). 

If you can share the real use-cases and related notation elements you are missing, I could share some more details how they should be placed in the tree of glyphs.
